manifests/charts/gateways/istio-ingress/templates/serviceaccount.yaml

```diff
@@ -10,6 +10,10 @@ imagePullSecrets:
 metadata:
   name: {{ $gateway.name | default "istio-ingressgateway" }}-service-account
   namespace: {{ .Release.Namespace }}
+  {{- if $gateway.serviceAccountRoleArn }}
+  annotations:
+    eks.amazonaws.com/role-arn: "{{ $gateway.serviceAccountRoleArn }}"
```

Comment:
Adding an annotation on the SA seems reasonable, but not sure we want an EKS specific one. I think other vendors have similar, so we don't want to end up with a bunch of different settings here

Reply:
I can refactor to add custom annotations on service account. This will resolve this problem without being cloud specific.

Reply:
FYI, the annotation can also be override by advanced k8s settings now, see https://github.com/istio/istio/tree/master/operator#advanced-k8s-resource-overlays